src/platform/platform_v4.cpp

```diff
@@ -155,13 +155,26 @@ namespace
 
             // Release all other CLR resources
             (void)metahost->Release();
-            (void)runtimeInfo->Release();
+
 
             // Start the runtime
             hr = runtimeHost->Start();
             IF_FAILURE_RETURN_OR_ABORT(ret, failure_load_runtime, hr, &_prepare_lock);
 
-            (void)runtimeHost->GetCurrentAppDomainId(&_appDomainId);
+            hr = runtimeHost->GetCurrentAppDomainId(&_appDomainId);
+			if (hr != S_OK)
+			{
+				ICorRuntimeHost* oldRuntimeHost;
+				hr = runtimeInfo->GetInterface(CLSID_CorRuntimeHost, IID_ICorRuntimeHost, (void**)&oldRuntimeHost);
+				IF_FAILURE_RETURN_OR_ABORT(ret, failure_get_current_app_domain_id_backup_try, hr, &_prepare_lock);
+				IUnknown* pUnk2;
+				hr = oldRuntimeHost->GetDefaultDomain(&pUnk2);
+				IF_FAILURE_RETURN_OR_ABORT(ret, failure_get_current_app_domain_id_backup_try, hr, &_prepare_lock);
+				hr = runtimeHost->GetCurrentAppDomainId(&_appDomainId);
+			}
+			(void)runtimeInfo->Release();//can't release earlierincase backup failure
+
+			IF_FAILURE_RETURN_OR_ABORT(ret, failure_get_current_app_domain_id, hr, &_prepare_lock);
```

Comment:
```suggestion
            hr = runtimeHost->GetCurrentAppDomainId(&_appDomainId);
	    if (hr != S_OK)
            {
                ICorRuntimeHost* oldRuntimeHost;
                hr = runtimeInfo->GetInterface(CLSID_CorRuntimeHost, IID_ICorRuntimeHost, (void**)&oldRuntimeHost);
                IF_FAILURE_RETURN_OR_ABORT(ret, failure_load_runtime, hr, &_prepare_lock);
                IUnknown* pUnk2;
                hr = oldRuntimeHost->GetDefaultDomain(&pUnk2);
                IF_FAILURE_RETURN_OR_ABORT(ret, failure_load_runtime, hr, &_prepare_lock);
                hr = runtimeHost->GetCurrentAppDomainId(&_appDomainId);
            }
            (void)runtimeInfo->Release(); //can't release earlier incase backup failure

            IF_FAILURE_RETURN_OR_ABORT(ret, failure_load_runtime, hr, &_prepare_lock);
```

I prefer spaces to tabs. I also updated the enum failure more and added a space in the comment.